File: Algorithms/Java/BinarySearch.java
class BinarySearch {
    int searchPosition(final int[] arr, final int target) {
        int left = 0;
        int right = arr.length - 1;

        while (left <= right) {
            int middle = (right - left) / 2 + left;
            if (arr[middle] == target) {
                return middle;
            } else if (arr[middle] < target) {
                left = middle + 1;
            } else if (arr[middle] > target) {
                right = middle - 1;
            }
        }
        return -1;
    }

    public static void main(final String[] args) {
        BinarySearch bs = new BinarySearch();
        int[] arr = { 1, 2, 3, 4, 5 };
        System.out.println(bs.searchPosition(arr, 3));
    }
}

File: Algorithms/Python/binary_search.py
from __future__ import annotations


def binary_search(arr: list[int], target: int) -> int:
    """
    Binary search algorithm to find the index of the target value in the array.
    If the target value is not in the array, return -1.
    """
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = (left + right) // 2

        if arr[mid] == target:
            return mid
        if arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1


if __name__ == "__main__":
    numbers = [1, 2, 3, 4, 5, 6]
    print(binary_search(numbers, 3))  # 2
    print(binary_search(numbers, 7))  # -1
